### PR TITLE
AWS EKS kubernetes resources

### LINKED ISSUE #143: Setup kubernetes cluster for explorer and workloads
```[tasklist]
### Tasks
- [x] EKS managed cluster for Explorers
- [x] Rancher Kubernetes for our networks
- [x] terraform declarations
```

### PR DESCRIPTION
**Pull request by DaMandal0rian**
The PR adds terraform declarations for AWS EKS managed kubernetes clusters on AWS infrastructure. 

- Cluster and node pool resources are created
- IAM role and KMS key management
- Bottlerocket optimized cluster with custom AMI for better performance and security
- gitops with argocd
- enable coredns, prometheus, cert-manager addons

closes #143

### CODE REVIEW

File: eks/base/eks_managed_node_group/main.tf [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+        # This could leave you with no way to access the API and change settings on an existing node!
+        [settings.host-containers.control]
+        enabled = true
+
+        # extra args added
+        [settings.kernel]
+        lockdown = "integrity"
+
+        [settings.kubernetes.node-labels]
+        # todo: add label to the nodes
```
Comment (vedhavyas) [inline eks/base/eks_managed_node_group/main.tf]: can we not get them through config and not hardcode it here ?

Comment (DaMandal0rian) [reply eks/base/eks_managed_node_group/main.tf]: yeah done

File: eks/base/eks_managed_node_group/variables.tf [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  #   },
+  # ]
+}
+
+variable "aws_auth_users" {
+  type        = list(any)
+  description = "List of AWS IAM accounts to add to aws-auth configmap"
+  default     = []
+
+  # i.e
```
Comment (vedhavyas) [inline eks/base/eks_managed_node_group/variables.tf]: do we need these commented out default ?

File: eks/base/eks_managed_node_group/variables.tf [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        [settings.host-containers.control]
+        enabled = true
+
+        # extra args added
+        [settings.kernel]
+        lockdown = "integrity"
+
+        [settings.kubernetes.node-labels]
+        # todo: add label to the nodes
+        app1 = "explorer-squid"
```
Comment (vedhavyas) [inline eks/base/eks_managed_node_group/variables.tf]: I still see these applications hard coded. Can we get these from config instead?

**Review by vedhavyas — COMMENTED** [review]
Make sense. Left some suggestions

File: eks/base/eks_managed_node_group/variables.tf (hunk 3 above)
Comment (DaMandal0rian) [reply eks/base/eks_managed_node_group/variables.tf]: sure, I have refactored it now, created blue/green environments and added argocd too. These hardcoded values are passed as input variables now.

**Review by vedhavyas — APPROVED** [review]

**Pull request merged**